factory/bootstrap/bootstrapComponents.go

```diff
@@ -157,6 +163,11 @@ func (bcf *bootstrapComponentsFactory) Create() (*bootstrapComponents, error) {
 		return nil, err
 	}
 
+	bootstrapHeartbeatSender, err := bcf.createHeartbeatSender(genesisShardCoordinator)
```

Comment:
we need to move this **after** the shard coordinator is assembled (L234)
Reason: at this point, the shard coordinator has the selfShardID value set to common.MetachainID (you can easily test this by taking this binary and trying to connect to testnet)

Reply:
done, moved just before sync start